account_cash_discount_base/models/account_invoice.py

```diff
@@ -98,6 +98,14 @@ def _compute_has_discount(self):
                 rec.type == 'in_invoice'
             )
 
+    @api.model
+    def _get_discount_base_date(self):
+        discount_base_date = datetime.today()
+        if self.date_invoice:
+            discount_base_date = \
+                fields.Date.from_string(self.date_invoice)
+        return discount_base_date
```

Comment:
I think this must be a computed field, not a method, so the onchange below can trigger when it changes.

Otherwise when you will override (to depend on the reception date), the onchange will not trigger when reception date changes.

Reply:
If that does not work, we close this, and override the whole onchange in the custom module.

@benwillig what do you think?

Reply:
If we have a computed field we can't use ```datetime.today()``` or it will change every day (or maybe it would be a stored field?). Then we would need to use the ```create_date``` as reference, but it's available only after the record has been created. But yes, it might be usefull to have a specific field which defines the base date.

Reply:
If the field is stored, yes.

Reply:
Since this field would be used only in onchange it should not matter